tests/pragmas/tcustom_pragma.nim

```diff
@@ -233,3 +233,10 @@ block:
   doAssert ps.first == ps[0] and ps.first == "one"
   doAssert ps.second == ps[1] and ps.second == 2
   doAssert ps.third == ps[2] and ps.third == 3.0
+
+# pragma with implicit&explicit generic types
+block:
+  template fooBar[T](x: T; c: static[int] = 42; m: char) {.pragma.}
```

Comment:
What is the difference of ``static[int]`` vs ``int`` in this context?

Reply:
- It introduces an implicit generic, this is useful to test the patch works as intended.
- I use `static[int]` because I need a literal value

Reply:
I can imagine that at some point I will at a check during sematic checking that isallows ``static[T]`` in the context of pragmas entirely. Simply because I don't see the advantage to support this feature. Then the test will be stripped of the ``static``. It will then of course still run, but it won't test your generic that you want.

So is it possible to write a test that does crash on current devel that does not use ``static`` at all?

Reply:
Yeah let's fucking remove stuff we don't like. What a fucking circus.

Reply:
Sorry? Yes I like to remove things if they meet the following conditions:

 * The implementation has bugs
 * The implementation isn't used much (hard to measure)
 * The language already allows an alternative solution that seems to be much more stable.
 * We are still pre version 1.0

I personally think this is a problem of the past of Nim, where features were added to the language without properly exploring the features that are already existing, and without properly looking into the consequences of that feature (bugs when combined with the other forgotten features). If your have problems with this plan, then please elaborate. Because I generally like your influence on the language.

Reply:
It's wrong to assume that `static` is not useful in this context. I think it's perfectly legal for a custom pragma to accept a run-time expression of a particular type. This may be useful when indicating the default value of something for example:

```nim
type Foo = object
   somePath {.defaultValue: getCurrentDir().}
```

This works fine, because the custom pragma value can be manipulated as a regular AST. Another pragma that doesn't support such run-time expression would naturally indicate this with a static parameter.